src/index.js

```diff
@@ -368,6 +370,110 @@ export class NetworkInfo {
 		this::_ensureDefaultOptions();
 	}
 
+	async applySettings (name, settings) {
+		if (_isNullOrUndefined(name)) {
+			throw new Error('interface name is required');
+		}
+
+		if (_isNullOrUndefined(settings)) {
+			throw new Error('settings are required');
+		}
+
+		// get the interface... if it does not exist, this will result in error
+		let
+			iface = await this::_ifconfig(name),
+			result;
+
+		// check to see if adapter should be enabled / disabled
+		if (!_isNullOrUndefined(settings.active)) {
+			debug(
+				'attempt %sabling interface %s',
+				settings.active ? 'en' : 'dis',
+				name);
+
+			if (settings.active) {
```

Comment:
do you think we should check if the interface is already enabled before enabling it again?

Reply:
I thought about that, but because of the nature of `ifconfig` and how it operates as a tool, there are no side effects (i.e. calling `ifconfig en0 up` on `en0` when it is already up does not create a problem)

While trying to strike the right balance between making the library smart and protect the user, generally speaking, I'm erring towards simplicity whenever possible.